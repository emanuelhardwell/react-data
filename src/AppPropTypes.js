import React from "react";
import PropTypes from "prop-types";

const AppPropTypes = ({
  saludo = "Saludo secundario en caso de que no se mande ninguno desde el padre",
}) => {
  return (
    <div>
      <h1> hola mundo </h1>

      <h2> Saludo: </h2>
      <h4> {saludo} </h4>
      <p> Estos son los PropTypes </p>
    </div>
  );
};

AppPropTypes.propTypes = {
  saludo: PropTypes.string.isRequired,
};

export default AppPropTypes;
