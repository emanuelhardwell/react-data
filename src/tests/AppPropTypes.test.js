//
//

// import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AppPropTypes from "../AppPropTypes";

describe("Pruebas en el archivo <AppPropTypes.js/> ", () => {
  //   test("should decir el saludo Hola, soy goku", () => {
  //     const saludo = "Hola, soy goku";
  //     // const wrapper = render(<AppPropTypes />);
  //     // expect(wrapper.getByText(saludo)).toBeInTheDocument();
  //     // const { getByText } = render(<AppPropTypes saludo={saludo} />);
  //     // expect(getByText(saludo)).toBeInTheDocument();
  //   });

  test("should de mostrar <AppPropTypes.js/> correctamente", () => {
    const saludo = "Hola, soy goku";
    const wrapper = shallow(<AppPropTypes saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });
});
