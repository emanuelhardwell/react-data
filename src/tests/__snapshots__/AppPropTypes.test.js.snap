// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <AppPropTypes.js/>  should de mostrar <AppPropTypes.js/> correctamente 1`] = `
<div>
  <h1>
     hola mundo 
  </h1>
  <h2>
     Saludo: 
  </h2>
  <h4>
     
    Hola, soy goku
     
  </h4>
  <p>
     Estos son los PropTypes 
  </p>
</div>
`;
